partiql-lib/partiql-sprout/src/main/kotlin/org/partiql/sprout/model/Model.kt

```diff
@@ -27,28 +27,34 @@ class Universe(
 /**
  * Definition of some type
  */
-sealed class TypeDef(
-    val ref: TypeRef.Path,
-) {
+sealed class TypeDef(val ref: TypeRef.Path) {
 
     /**
-     * Children represent type definitions nested within another, but not related algebraic-ly.
-     * That is, it's a convenience
+     * All types defined within this node
      */
     abstract val children: List<TypeDef>
 
+    /**
+     * Types defined within this node which are not related to this node "algebraically" such as inline definitions
+     */
+    abstract val types: List<TypeDef>
+
     /**
      * TypeDef.Sum represents a list of type variants
      */
-    class Sum(ref: TypeRef.Path, val variants: List<TypeDef>, override val children: List<TypeDef>) : TypeDef(ref) {
+    class Sum(ref: TypeRef.Path, val variants: List<TypeDef>, override val types: List<TypeDef>) : TypeDef(ref) {
+
+        override val children: List<TypeDef> = variants + types
 
         override fun toString() = "sum::$ref"
     }
 
     /**
      * TypeDef.Product represents a structure of name/value pairs
      */
-    class Product(ref: TypeRef.Path, val props: List<TypeProp>, override val children: List<TypeDef>) : TypeDef(ref) {
+    class Product(ref: TypeRef.Path, val props: List<TypeProp>, override val types: List<TypeDef>) : TypeDef(ref) {
+
+        override val children: List<TypeDef> = props.filterIsInstance<TypeProp.Inline>().map { it.def } + types
```

Comment:
Confused here. is this a mismatch between the `children` in the core model and `children` in `KotlinNode`?

Reply:
You're right. There is a slight mismatch. This model keeps inline definitions within the TypeProp.Inline variant whereas the Kotlin generation separates them out. For the model, keeping inline definitions within TypeProp captures the fact that these definitions are inline nested definitions rather than normal nested definitions. For Kotlin generation, the distinction doesn't matter hence why it's dropped in the generator, but I figured the distinction was worth keeping in the model.